### DISCUSSION TITLE
Qucs-S 24.3.0

### DISCUSSION THREAD

**Discussion by ra3xdh** [ra3xdh/qucs_s#860] [Announcements]
# Qucs-S 24.3.0

## New features

* SpiceLibComp redesign. This device is renamed to *Spice Library Device* and provides an easy way 
  to add SPICE model to schematic and assign symbol to it. See #679
* Added possibity to import SPICE model for Diode, BJT, MOSFET, and JFET unified devices (blue) #795
* Implemeted symbol files (`*.sym`)  editing mode. These files are needed to attach symbol to SPICE devices #700
* Redesign of schematic rendering engine. Improve rendering quality on HiDPI displays #723
* Improved Qucsconv GUI to reflect new converter features #826
* Improved libraries portability, relative path support for libraries, show libraries from project directory #567
* Added a global setting for Ngspice compatibility mode (LTspice, HSPICE, etc.) #851
* Add global Ngspice compatibility mode (LTspice/HSPICE/S3) setting #851
* QucsatorRF updated to version 1.0.1

## Deprecated features

* The `*.qucs` binary archives are not supported anymore. Use system archivers to pack the projects instead #811
* The `SweepModel` property of *Parameter Sweep* simulaiton has been deprecated. It is not needed anymore after Ngspice
  has added variable sweep feature. Use variable sweep instead #839
* Toolbars and status bar made always visible #852
* Icons theme settings removed from settings dialog. These settings have no effect since v2.1.0
  when unified and auto-generated icons were introduced.

## Packaging

* Linux DEB and RPM packages switched to Qt6
* Windows package switched to Qt6
* Introduced continous release of Linux, Windows, and Mac packages #719 
* Added MacOS package build #797 and homebrew support #804

## General improvements

* The TEMP and TEMPER variables are recognized by parameter sweep to apply temperature sweep #816
* Fixed ASCO optimization with QucsatorRF #830
* Enabled variables as parameters for unified nonlinear devices and pulsed source #838 and #808
* Fixed issues with XYCE simulation #701
* Improved attenuator synthesis tool #722
* Improved filter synthesis tool #715 #673
* Redesign *File->Open examples* menu. Show open dialog instead of file manager #672
* Qucs-S now follows XDG standard for cached data location #145
* Apply .OPTION INTERP when simulating transient with parameter sweep; prevent broken par. sweep data  #766
* Implemented relative paths support for library devices #567

## Component library

* Added PWM controllers (TL494, UC384x, etc.) library #846
* Added Germanium diodes library #846
* Added Mixer IC library containing SA612 model #846
* Added Diodes extended library containing different USSR diodes models
* Added new Varactor library #759
* Added new PIN diode library #764

## Localization

* Russian translation update

# Installation

See the wiki page for the installation instructions: https://github.com/ra3xdh/qucs_s/wiki/Installation

<hr /><em>This discussion was created from the release <a href='https://github.com/ra3xdh/qucs_s/releases/tag/'>Qucs-S 24.3.0</a>.</em>

**Comment by dsm** [comment]
Hi, Version is wrong for windows installer exe. version 24.1.0 and qt version 6.7.0 but our snapshot build 6.7.2.

**Reply by ra3xdh** [reply]
Windows installer is build manually, because it contains ngspice and asco versions for Windows. My local MSYS2 installation may be not synchronized with the latest Qt update. The MSYS is quite fragile and may be broken after libraries update. The version of the application should be correct. I will check it again.

**Reply by ra3xdh** [reply]
Yes, I didn't change the version inside InnoSetup script. I will reupload the installer today. The version of the Qucs-S itself inside is correct and is 24.3.0.

**Reply by dsm** [reply]
Yes, just installer title wrong. I think, we can improve our CI to release .exe via github for next release. if any tag pushed repo deploy.yml run and create release for this tag.

**Reply by dsm** [reply]
https://github.com/Minionguyjpro/Inno-Setup-Action

**Reply by dsm** [reply]
can you add macos dmg files to release.

**Reply by ra3xdh** [reply]
Yes, I will upload them later today or tomorrow.

**Reply by dsm** [reply]
I tried to download macos arm64 file but I couldn't download, intel one is ok, and dmg file shouldn't be zip because I am planing add stable release to brew

**Reply by ra3xdh** [reply]
Then DMG packages are taken from here: https://github.com/ra3xdh/qucs_s/actions/runs/10054938118 I am wondering why Github didn't upload ARM version properly.

**Reply by ra3xdh** [reply]
I have replaced the Windows installler file.

**Reply by dsm** [reply]
can you change macos release to unzip version.

**Reply by ra3xdh** [reply]
I have just uploaded unzipped DMG packages.

**Comment by ra3xdh** [comment]
Linux DEB/RPM installation is possible only using *Grab package directly* option. The insttallation from RPM repository fails because of checksum error, and the installation from DEB fails because of file size mismatch and says that the mirror is syncing. See #872. The OBS doesn't provide any means to debug this process. The help from OBS experts is highly welcome.

**Reply by ra3xdh** [reply]
The installation from DEB repository has been fixed.

**Reply by ra3xdh** [reply]
The installation of RPM packages has been self-fixed. The adding repository on fresh VM works as expected.
